ctapipe/io/hdf5eventsource.py

```diff
@@ -272,16 +287,32 @@ class ObsIdContainer(Container):
             default_prefix = ""
             obs_id = Field(-1)
 
-        simulation_configs = {}
         if "simulation" in self.file_.root.configuration:
             reader = HDF5TableReader(self.file_).read(
                 "/configuration/simulation/run",
                 containers=(SimulationConfigContainer, ObsIdContainer),
             )
-            for (config, index) in reader:
-                simulation_configs[index.obs_id] = config
+            return {index.obs_id: config for (config, index) in reader}
+        else:
+            return {}
+
+    def _parse_sb_and_ob_configs(self):
+        """read Observation and Scheduling block configurations"""
+
+        sb_reader = HDF5TableReader(self.file_).read(
+            "/configuration/observation/scheduling_block",
+            containers=SchedulingBlockContainer,
+        )
+
+        scheduling_blocks = {sb: sb.sb_id for sb in sb_reader}
```

Comment:
This dict comprehension looks switched. Shouldn't it be: `{sb.sb_id: sb ...}`

Reply:
Yup, clearly wrong